src/modules/location/index.ts

```diff
@@ -88,18 +112,20 @@ export class LocationModule {
           state?: string;
         } = {}
   ): string {
+    deprecated({
+      deprecated: 'faker.location.zipCodeByState',
+      proposed: 'faker.location.zipCode({ state })',
+      since: '8.0',
+      until: '9.0',
+    });
+
     if (typeof options === 'string') {
       options = { state: options };
     }
 
     const { state } = options;
 
-    const zipRange = this.faker.definitions.location.postcode_by_state?.[state];
-    if (zipRange) {
-      return String(this.faker.number.int(zipRange));
-    }
-
-    return this.zipCode();
+    return this.zipCode({ state });
```

Comment:
IMO there is no need to repackage the options, just pass them along as is.

```suggestion
    return this.zipCode(options);
```

Reply:
How important is this change to you and do you have a reason for passing the whole options instead only what is needed?

Reply:
Because it is the same, and even if not, it doesn't matter either.
It is not important to me though.